src/dodal/devices/utils.py

```diff
@@ -12,3 +18,84 @@ def epics_signal_put_wait(pv_name: str, wait: float = 1.0) -> EpicsSignal:
         EpicsSignal: An EpicsSignal that will wait for a callback.
     """
     return Component(EpicsSignal, pv_name, put_complete=True, write_timeout=wait)
+
+
+class StatusException(Exception):
+    """For general status failures"""
+
+    pass
+
+
+def wrap_and_do_funcs(
+    unwrapped_funcs: list[Callable[[], StatusBase]],
+    timeout: float = 60.0,
+) -> Status:
+    """Creates and initiates an asynchronous chaining of functions which return a status.
+
+    Usage:
+    This function can be used to convert a series of blocking, status-returning functions to a series of sequential, asynchronous,
+    status-returning functions by making use of callbacks. It also checks for exceptions on each returned status
```

Comment:
Nit: Still doesn't sound quite right.
```suggestion
    This function can be used to take a series of status-returning functions and run them all sequentially and in the background by making use of callbacks. It also ensures exceptions on each returned status are propagated
```